modules/nw-ingress-controller-nodeportservice-projects.adoc

```diff
@@ -0,0 +1,136 @@
+// Module included in the following assemblies:
+//
+// * networking/configuring_ingress_cluster_traffic/nw-configuring-ingress-controller-endpoint-publishing-strategy.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="nw-ingress-controller-nodeportservice-projects_{context}"]
+= Adding a single NodePort service to an Ingress Controller
+
+Instead of creating a `NodePort`-type `Service` for each project, you can create an Ingress Controller to use the `NodePortService` endpoint publishing strategy. To prevent port conflicts, consider this configuration for your Ingress Controller when you want to apply a set of routes, through Ingress sharding, to nodes that might already have a `HostNetwork` Ingress Controller.  
+
+Before you `NodePort`-type `Service` for each project, read the following considerations:
+
+* You must create a wildcard DNS record for the Ingress Controller.
+* You must expose a route for your service and specify the `--hostname` argument for your Ingress Controller domain.
+* must preserve the non-default port in the curl in order to access, for example: curl <route>:<port> // what does this mean?
+* Ensure that Ingress Controller routes can access your {product-title} cluster through the host IP address assigned to each node.
+
+.Prerequisites
+
+* You installed the {oc-first}.
+* Logged in as a user with `cluster-admin` privileges.
+* You created a wildcard DNS record.
+// https://docs.openshift.com/container-platform/4.16/networking/ingress-controller-dnsmgt.html (does not detail how to create the DNS)
+
+.Procedure
+
+. Create a custom resource (CR) file for the Ingress Controller:
++
+.Example of a CR file that defines information for the Ingress Controller
+[source,yaml]
+----
+apiVersion: v1
+items:
+- apiVersion: operator.openshift.io/v1
+  kind: IngressController
+  metadata:
+    name: custom-ic3 <1>
+    namespace: openshift-ingress-operator
+  spec:
+    replicas: 1
+    domain: my-custom-ic.ipi-cluster.example.com <2>
+    nodePlacement:
+            nodeSelector:
+              matchLabels:
+                ingresscontroller: "custom-ic3" <3>
+    namespaceSelector:
+     matchLabels:
+       ic: custom-ic3 <4>
+    endpointPublishingStrategy:
+      type: NodePortService 
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+# ...
+----
+<1> Specify the `name` with a name for the `IngressController` object.
+<2> Specify the `domain` for the wilcard DNS record.
+<3> Specify the label for the `ingresscontroller` CR.
+<4> Matches routes from any namespaces with this label.
+
+. Add a label to a node by using the `oc label node` command:
++
+[source,terminal]
+----
+$ oc label node <node_name> ingresscontroller=<ingress_controller_custom_resource>
+----
+
+. Create the `Pod` object:
++
+[source,terminal]
+----
+$ oc create -f <pod_object_name>.yaml
```

Comment:
```suggestion
$ oc create -f <ingresscontroller>.yaml
```
This is the file where the Ingress Controller yaml went.

Reply:
:heavy_check_mark: